### PR TITLE
[forge] Split out test suites from main

### PR DESCRIPTION
**Pull request by perryjrandall**
There are a lot of tests and cruft in with the main forge cli

Split them out into suites

Test Plan:

running forge on PR succeeds

### CODE REVIEW

**Review by aluon — APPROVED** [review]
LGTM

File: testsuite/forge-cli/src/suites/mod.rs [hunk 1]
```diff
@@ -1,4 +1,15 @@
 // Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
 pub mod dag;
+pub mod db;
```
Comment (aluon) [inline testsuite/forge-cli/src/suites/mod.rs:6]: Probably for a future PR. Now that we have the test suites split up into different modules, it'd be great to have descriptions for what each test suite is doing. @igor-aptos added some comments for forge-stable in #14775 , maybe we should move some of the docs to these files?

**Review by banool — APPROVED** [review]

File: testsuite/forge-cli/src/main.rs [hunk 2]
```diff
@@ -617,7 +518,7 @@ fn get_test_suite(
 
     // Otherwise, check the test name against the grouped test suites
     if let Some(test_suite) = get_land_blocking_test(test_name, duration, test_cmd) {
-        return Ok(test_suite);
+        Ok(test_suite)
```
[the remark this answers is not in the document]
Comment (perryjrandall) [reply testsuite/forge-cli/src/main.rs:519]: Woops yeah i already fixed this in another commit forgot to pick

File: testsuite/forge-cli/src/suites/mod.rs (hunk 1 above)
Comment (perryjrandall) [reply testsuite/forge-cli/src/suites/mod.rs:6]: Yeah in general i think these should live with the tests themselves not in docs on a forge workflow

**Pull request merged**